### PR TITLE
Use ES6 class for react components

### PR DESCRIPTION
**Pull request by Holi0317**
Note: this commit will disable eslint rule, `no-invalid-this`. Because the following code will fail this rule:
```js
class Comp extends React.Component {
  myCallback = e => {
      // Do stuff
  }
}
```

### CODE REVIEW

**Comment by Andarist** [comment]
You can enable this rule again, but at the same time u need to change those lines:
https://github.com/Andarist/react-collapse/blob/badc37ddc3f377f84c1291effbe8e08acf42ad5f/.eslintrc#L10
https://github.com/Andarist/react-collapse/blob/badc37ddc3f377f84c1291effbe8e08acf42ad5f/.eslintrc#L107

+ ofc `npm install --save-dev eslint-plugin-babel`

Also peer dependency should be changed to - `"react": ">=15.3"`, because of the `PureComponent`

File: package.json [hunk 1]
```diff
@@ -66,9 +67,9 @@
     "json-loader": "0.5.4",
     "nightwatch-autorun": "3.1.0",
     "parallelshell": "2.0.0",
+    "react": "15.5.4",
```
Comment (Andarist) [inline package.json]: Also peer dependency should be changed to - `"react": ">=15.3"`, because of the `PureComponent`

File: .eslintrc [hunk 2]
```diff
@@ -104,7 +104,6 @@
     "no-implicit-coercion": 2, // disallow the type conversions with shorter notations
     "no-implicit-globals": 2, // disallow var and named functions in global scope
     "no-implied-eval": 2, // disallow use of eval()-like methods
-    "no-invalid-this": 2, // disallow this keywords outside of classes or class-like objects
```
Comment (Andarist) [inline .eslintrc]: You can enable this rule again, but at the same time u need to change those lines:
https://github.com/Andarist/react-collapse/blob/badc37ddc3f377f84c1291effbe8e08acf42ad5f/.eslintrc#L10
https://github.com/Andarist/react-collapse/blob/badc37ddc3f377f84c1291effbe8e08acf42ad5f/.eslintrc#L107

plus ofc `npm install --save-dev eslint-plugin-babel`

**Comment by nkbt** [comment]
Yeah, that's ok. The only downside is that I will need to publish major release since React 0.14 support is gone. Not like someone actually using it, but semver is cheap anyway...

**Comment by Andarist** [comment]
Hm. 2.x branch should be bumped then too, but to what :trollface:?

**Comment by nkbt** [comment]
Yeah... would have to break semver for it :(. It is sort of deprecated anyway.  Will add a note in readme about it

**Comment by Holi0317** [comment]
Fixed those issues. Thanks @Andarist

**Pull request merged**